File: go_backend/scripts/migrate.go
package main

import (
	"fmt"
	"os"

	"syncflow/config"
	"syncflow/database"
	"syncflow/logger"
	"go.uber.org/zap"
)

func main() {
	cfg, err := config.LoadConfig()
	if err != nil {
		fmt.Printf("Failed to load config: %v\n", err)
		os.Exit(1)
	}

	logger.Init(cfg.Environment)
	defer logger.Sync()

	logger.Info("Running standalone migration...")

	db, err := database.InitDB(cfg)
	if err != nil {
		logger.Fatal("Could not connect to database", zap.Error(err))
	}

	if err := database.RunMigrations(db); err != nil {
		logger.Fatal("Migration failed", zap.Error(err))
	}

	logger.Info("Migration finished successfully")
}

File: node_backend/src/scripts/migrate.ts
// scripts/migrate.ts — Equivalent to go_backend/scripts/migrate.go
import { loadConfig } from '../config/config';
import { initDB, disconnectDB } from '../database/connection';
import { runMigrations } from '../database/migration';
import logger from '../logger/logger';

/**
 * Standalone migration script.
 * Run with: npm run migrate
 */
async function main(): Promise<void> {
    const cfg = loadConfig();

    logger.init(cfg.environment);
    logger.info('Running standalone migration...');

    // Initialize DB to validate connection
    initDB(cfg);

    try {
        await runMigrations(cfg.environment);
        logger.info('Migration finished successfully');
    } catch (err) {
        logger.error('Migration failed', {
            error: err instanceof Error ? err.message : String(err),
        });
        process.exit(1);
    } finally {
        await disconnectDB();
    }
}

main();
